analysis/test_data_utils.py

```diff
@@ -69,6 +70,42 @@ def test_drop_uniteresting_columns():
     assert 'time_started' not in cleaned_df.columns
 
 
+def test_clobber_snapshots():
+    """Tests that clobber snapshots clobbers stale snapshots from earlier
+    experiments."""
+    experiments = []
+    df = pd.DataFrame()
+    for experiment_num in range(3):
+        experiment = 'experiment-%d' % experiment_num
+        experiments.append(experiment)
+        experiment_df = create_experiment_data(experiment)
+        df = pd.concat([df, experiment_df])
+    not_updated_benchmark = 'libpng'
+    not_updated_fuzzer = 'afl'
+    drop_condition = ((df.experiment != experiments[2]) |
+                      (df.benchmark != not_updated_benchmark) |
+                      (df.fuzzer != not_updated_fuzzer))
+    df = df[drop_condition]
+    df = data_utils.clobber_snapshots(df, experiments)
+    # All of the first experment should be clobbered
```

Comment:
This test feels like a bit convoluted way of checking the results. Consider just checking if df[['experiemnt', 'benchmark', 'fuzzer']].drop_duplicates() is the expected 4 triples, eg.:
```
'experiment1', 'libxml', 'afl'
'experiment2', 'libxml', 'libfuzzer'
'experiment2', ...
'experiment2', ...
```